diff --git a/gst/rtsp/gstrtspsrc.c b/gst/rtsp/gstrtspsrc.c
index 168c5e30d..262b35c1f 100644
--- a/gst/rtsp/gstrtspsrc.c
+++ b/gst/rtsp/gstrtspsrc.c
@@ -2894,16 +2894,23 @@ gst_rtspsrc_handle_src_sink_event (GstPad * pad, GstObject * parent,
     GstEvent * event)
 {
   GstRTSPStream *stream;
+  GstRTSPSrc *self = GST_RTSPSRC (GST_OBJECT_PARENT (parent));
 
   stream = gst_pad_get_element_private (pad);
 
   switch (GST_EVENT_TYPE (event)) {
     case GST_EVENT_STREAM_START:{
-      const gchar *upstream_id;
+      GChecksum *cs;
+      gchar *uri;
       gchar *stream_id;
 
-      gst_event_parse_stream_start (event, &upstream_id);
-      stream_id = g_strdup_printf ("%s/%s", upstream_id, stream->stream_id);
+      cs = g_checksum_new (G_CHECKSUM_SHA256);
+      uri = self->conninfo.location;
+      g_checksum_update (cs, (const guchar *) uri, strlen (uri));
+      stream_id =
+          g_strdup_printf ("%s/%s", g_checksum_get_string (cs),
+          stream->stream_id);
+      g_checksum_free (cs);
 
       gst_event_unref (event);
       event = gst_event_new_stream_start (stream_id);
